packages/core/src/controllers/pairing.ts

```diff
@@ -111,15 +114,38 @@ export class Pairing implements IPairing {
 
   public pair: IPairing["pair"] = async (params) => {
     this.isInitialized();
-    this.isValidPair(params);
+
+    const event = this.core.eventClient.createEvent({
+      properties: {
+        topic: params?.uri,
+        trace: [EVENT_CLIENT_PAIRING_TRACES.pairing_started],
+      },
+    });
+
+    try {
+      this.isValidPair(params, event);
+    } catch (error) {
+      event.setError(EVENT_CLIENT_PAIRING_ERRORS.malformed_pairing_uri);
```

Comment:
Wouldn't this end up in a duplicated event since `isValidPair` already takes care of setting the error event ?